[Damien Wee] iP

damiwee opened the pull request.

damiwee committed:
Feature: Add greet and exit messages

damiwee committed:
Feature: Add echo user command

damiwee committed:
Feature: Add task list for user input

damiwee committed:
Feature: Add command to mark or unmark user input task as done

damiwee committed:
Style: Comply with coding standards

damiwee committed:
Style: Capitalize constant variable

damiwee committed:
Style: Fix variable camelcase naming standard

damiwee committed:
Feature: Add 3 types of tasks

The 3 task types are todo, deadline and event.

Todo is a subclass of task class and both deadline
and event are subclasses of todo class. Introduce
inheritance for code reusability and promoting a
hierarchical structure

damiwee committed:
Test: Pass output match test

damiwee committed:
Refactor: Delete unused method

damiwee committed:
Style: Fix spacing

damiwee committed:
Refactor: Extract echo method

sweijie24 commented on src/main/java/BossMan.java:
```diff
@@ -0,0 +1,147 @@
+        return userInput.trim().split("\\s+", 2);
+    }
+
+    private void handleMarkCommand(String commandArgs) {
```
The handleMarkCommand and handleUnmarkCommand seems to be handled similarly. Consider refactoring it into a separate method to avoid duplication

royaliceblue replied:
Agreed! Perhaps it might affect readability as the function names are very similar? Could consider refactoring into a single method!

sweijie24 commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,62 @@
+import java.util.List;
+
+public class TaskList {
+    private final String SEP = "____________________________________________________________";
```
Consider making the constant here "static" if it is not being modified across all instances

sweijie24 commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,62 @@
+            System.out.println("Nice! I've marked this task as done:");
+            echo(num);
+        } else {
+            System.out.println("Selected index out of range" + "\n" + SEP);
```
Consider also throwing an exception instead of just printing a message

sweijie24 commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,62 @@
+    public void printTasks() {
+        System.out.println("Todo List:");
+        int counter = 1;
+        for (Task t: TASKS) {
```
Consider using a traditional for-loop standard since you are using a counter anyways. This can help the code to be more readable and perhaps avoid further issues in the event of any future modifications

sweijie24 commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,62 @@
+        System.out.println(SEP);
+    }
+
+    public void markTask(int num)  {
```
Similarly, the markTask and unmarkTask can perhaps be refactored into 1 whole method

royaliceblue commented on src/main/java/BossMan.java:
```diff
@@ -0,0 +1,147 @@
+
+    public void greetUser() {
+        System.out.println(SEP + "\nHello! I'm BossMan");
+        System.out.println("What can I do for you?\n" + SEP);
```
Would it be better to print a line break using the System.lineSeparator() as the behavior of the \n depends on the platform? System.lineSeparator() works the same in all platforms! 🙂

royaliceblue commented on src/main/java/BossMan.java:
```diff
@@ -0,0 +1,147 @@
+    }
+
+    private void handleDeadlineCommand(String commandArgs) {
+        String[] deadlineArgParts = commandArgs.split("/by", 2);
```
Is it necessary to add 2 to the split function? Will it be better to declare a constant for it?

royaliceblue commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,19 @@
+public class Deadline extends Todo {
+    private final String TYPE_SYMBOL = "[D]";
```
Would it be better not to redeclare the TYPE_SYMBOL as Todo class already has this attribute? 

Perhaps can try to modify this variable in your Deadline constructor than redeclaring?

royaliceblue commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,20 @@
+public class Event extends Todo {
+    private final String TYPE_SYMBOL = "[E]";
+    private final String FROM;
+    private final String TO;
+    public Event(String task, String from, String to) {
```
Would it be clearer to have a linespace between TO variable and Event constructor to enhance clarity?

royaliceblue commented on src/main/java/Main.java:
```diff
@@ -0,0 +1,8 @@
+public class Main {
+    public static void main(String[] args) {
+        BossMan chatBot = new BossMan();
```
I like the way of abstraction here

royaliceblue reviewed (commented):
Very nice way of code abstraction and refactoring.
Just a few nits. Overall great code! 😄

damiwee committed:
Feature: Handle invalid todo and unknown command exception

damiwee committed:
Feature: Handle invalid deadline command exception

damiwee committed:
Feature: Handle invalid event command exception

damiwee committed:
Feature: Handle invalid mark/unmark task format

damiwee committed:
Merge branch 'branch-Level-5'

damiwee committed:
Refactor: Package files

Group files by types exception or task

damiwee committed:
Refactor: Add BossMan exception class

damiwee committed:
Refactor: Change inheritance for exceptions related to BossMan

damiwee committed:
Refactor: Shift BossMan exception class to correct folder

StanleyW00 commented on src/main/java/bossman/exceptions/commandexceptions/BossManExceptions.java:
```diff
@@ -0,0 +1,5 @@
+
+public class BossManExceptions extends Exception{
+    //No additional code needed
+}
```
Try to avoid using empty classes for your exception. Perhaps, add a few methods or a method in handling the exception. Same goes for the other empty exception classes.

StanleyW00 commented on src/main/java/bossman/BossMan.java:
```diff
@@ -0,0 +1,166 @@
+        TASK_LIST.markTask(number);
+    }
+
+    private void handleUnmarkCommand(String commandArgs) throws NumberFormatException{
```
There is no spacing between the exception and the curly bracket.
`private void handleUnmarkCommand(String commandArgs) throws NumberFormatException{`
Same goes for several methods with this error.

StanleyW00 commented on src/main/java/bossman/BossMan.java:
```diff
@@ -0,0 +1,166 @@
+    }
+
+    private void handleEventCommand(String commandArgs) throws InvalidEventCommandException{
+        String[] eventArgParts = commandArgs.split("/from", 2);
+        if (eventArgParts.length != 2 || eventArgParts[0].isBlank()){
+            throw new InvalidEventCommandException();
+        }
+        String description = eventArgParts[0];
+        String[] eventArgTimeParts = eventArgParts[1].split("/to", 2);
+        if (eventArgTimeParts.length != 2){
+            throw new InvalidEventCommandException();
+        }
+        String from = eventArgTimeParts[0];
+        String to = eventArgTimeParts[1];
+        Task eventTask = new Event(description, from, to);
+        TASK_LIST.addTask(eventTask);
+        echo(eventTask);
```
Perhaps add blank lines to separate groups of related statements for improving readibility. Same goes for other methods with no blank spaces between several statements.

StanleyW00 reviewed (commented):
Overall, good job on maintaining code quality and following the coding standards. There's just a few improvements that could be made.

damiwee committed:
Refactor: Move exception messages into respective exception classes

To avoid empty classes

damiwee committed:
Feature: Add delete task command

damiwee committed:
Feature: Handle list index out of bounds exception for delete

damiwee committed:
Feature: Add read and write tasks from text file

damiwee committed:
Merge branch 'branch-Level-6'

damiwee committed:
Feature: Handle file not found exception

Create new file if file not found

damiwee committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/bossman/BossMan.java
#	src/main/java/bossman/task/TaskList.java

damiwee committed:
Style: Fix type symbol string format

damiwee committed:
Refactor: Edit exception handling

damiwee committed:
Refactor: Extract ui, parser and command classes

damiwee committed:
Refactor: Edit refactored ui, parser and command classes

damiwee committed:
Feature: Parse local datetime object for deadline task

damiwee committed:
Feature: Add find command

Filter tasks by description

damiwee committed:
Merge pull request #1 from damiwee/branch-Level-8

Feature: Parse local datetime object for deadline task

damiwee committed:
Merge branch 'master' of https://github.com/damiwee/ip

damiwee committed:
Merge branch 'master' into branch-Level-9

# Conflicts:
#	data/BossMan.txt

damiwee committed:
Merge pull request #2 from damiwee/branch-Level-9

Branch level 9

damiwee committed:
Merge branch 'master' of https://github.com/damiwee/ip

damiwee committed:
Document: Add javadocs and comments

damiwee committed:
Document: Add javadocs for bossman class

damiwee committed:
Merge pull request #3 from damiwee/branch-A-JavaDoc

Document: Add javadocs for bossman class

damiwee committed:
Fix: Add save file after each command too instead of save at end only

Previously, the application saves to file at the end of the program.
Now, fix the application to also save to file after each command.

damiwee committed:
Update README.md

damiwee committed:
Update README.md

damiwee committed:
Update README.md

damiwee committed:
Update README.md
